[RPC x JIT] Integrate Python RPC TorchScript call with profiler

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36266 [RPC x JIT] Integrate Python RPC TorchScript call with profiler**

RecordFunction starts but never ends at an appropriate location. Might be recording wrong time numbers.

This is following up on 

https://github.com/pytorch/pytorch/pull/32197
and
https://github.com/pytorch/pytorch/pull/32466

Differential Revision: [D7857991](https://our.internmc.facebook.com/intern/diff/D7857991/)

xush6528 committed:
[RPC x JIT] Integrate Python RPC TorchScript call with profiler

RecordFunction start but never end at an appropriate location. Might be recording wrong time number.

Differential Revision: [D7857991](https://our.internmc.facebook.com/intern/diff/D7857991/)

[ghstack-poisoned]

rohan-varma commented on torch/csrc/distributed/rpc/init.cpp:
```diff
@@ -476,8 +477,9 @@ If the future completes with an error, an exception is thrown.
-         const py::args& args,
-         const py::kwargs& kwargs) {
+         const std::shared_ptr<torch::autograd::profiler::RecordFunction>&
+             rf const py::args& args,
```
This won't compile, right? Missing comma between rf and next arg?

rohan-varma reviewed (changes requested):
Don't we need to make the relevant API changes in `torchscript_functions.h`? Also, will adding the extra arg cause any errors with the JIT call in `register_distributed_ops`? I'm assuming that if we default RecordFunction to nullptr, then it should be fine we and we won't have to make any JIT changes.

xush6528 committed:
Update on "[RPC x JIT] Integrate Python RPC TorchScript call with profiler"


RecordFunction start but never end at an appropriate location. Might be recording wrong time number.

Differential Revision: [D7857991](https://our.internmc.facebook.com/intern/diff/D7857991/)

[ghstack-poisoned]
